### PR TITLE
Refactor to use DockerHub and new Recipe layout

### PR DESCRIPTION
**Pull request by ciaransweet**
## What I am changing

TL;DR: This PR refactors out the `images/` directory for the Prefect Agent & Worker images in favour of using the new 
`pangeo-forge-bakery-images` DockerHub repository. It also adds the `oisst_recipe.py` Recipe which the Azure Bakery uses as this is now the new go-to for Recipe -> Flow conversion. To match the Azure Bakery, I've also switched from `pipenv` to `poetry` - I was getting fed up of flakey locking on `pipenv` (Sorry @sharkinsspatial)

## How I did it

* Swapped out `pipenv` for `poetry`
* Now use a local `.venv` when using `poetry` (also makes caching on CI less gruesome)
* Updates to `README.md`:
  * Broke deployment into `deploying` and `destroying`
  * Fleshed out the `.env` explanation
  * Updated the setup instructions
  * Added note on setting up the variable `BAKERY_IMAGE`
  * General aligning against the Azure Bakery README
  * Notes on registering flows
* Update `cdk/bakery_stack.py` to use the `BAKERY_IMAGE` variable and to specifically invoke Prefect (instead of relying on the docker command)
* Fleshed out `example.env`
* Removed all unnecessary flows and images
* Adding `oisst_recipe.py` to mirror the one provided in the Azure Bakery. With obvious modifications to talk to AWS/Fargate/Dask
* General bump of CDK to `1.109.0`
* Tidy up of `Makefile` and adding `register-flow`

## How you can test it

Follow the setup instructions in `README.md` then:

```bash
$ make deploy # deploy the bakery
$ poetry run dotenv run python3 flow_test/oisst_recipe.py # register the flow with Prefect 
```

Then invoke the flow via Prefect, you should get a passing Flow run in ~12 minutes:

image

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,26 +1,35 @@
...
-	pipenv run npx cdk deploy --app cdk/app.py --require-approval never
+	poetry run dotenv run npx cdk deploy --app cdk/app.py --require-approval never
 
+.PHONY: destroy
 destroy:
-	pipenv run npx cdk destroy --app cdk/app.py --force
+	poetry run dotenv run npx cdk destroy --app cdk/app.py --force
+
+.PHONY: register-flow
+register-flow:
```
Comment (sharkinsspatial) [inline Makefile:29]: Much better approach here 👍

File: README.md [hunk 2]
```diff
@@ -8,18 +8,20 @@ This repository serves as the provider of an AWS CDK Application which deploys t
...
 
 ## Requirements
 
 To develop on this project, you should have the following installed:
 
-* Node 14 (We recommend using NVM [Node Version Manager](https://github.com/nvm-sh/nvm))
+* [Node 14](https://nodejs.org/en/) (We recommend using NVM [Node Version Manager](https://github.com/nvm-sh/nvm))
 * [AWS CDK](https://docs.aws.amazon.com/cdk/latest/guide/getting_started.html) - There is a `package.json` in the repository, it's recommended to run `npm install` in the repository root and make use of `npx <command>` rather than globally installing AWS CDK
-* Python 3.8.* (We recommend using [pyenv](https://github.com/pyenv/pyenv))
-* [pipenv](https://github.com/pypa/pipenv)
```
Comment (sharkinsspatial) [inline README.md:22]: 🎊

**Review by sharkinsspatial — APPROVED** [review]
